### PR TITLE
fix: Fix for syncing standalone sessions for AirBeamMini 

### PR DESCRIPTION
**Pull request by mpanuszewska**
Trello ticket: 
https://trello.com/c/4LjP33lu/902-1160-4-abm-sync-doesnt-work

### CODE REVIEW

File: AirCasting/CoreData/MeasurementStream.swift [hunk 1]
```diff
@@ -9,6 +9,8 @@ enum MeasurementStreamSensorName: String {
     case pm1 = "AirBeam3-PM1"
     case pm2_5 = "AirBeam3-PM2.5"
     case pm10 = "AirBeam3-PM10"
+    case mini_pm1 = "AirBeamMini-PM1"
```
Comment (mpanuszewska) [inline AirCasting/CoreData/MeasurementStream.swift:12]: Airbeam Mini is technically a different sensor so we have to have it separate from the AB3 sensors, even if they are measuring the same thing.

File: AirCasting/SDSync/Parsing/MiniSDCardMeasurementsParser.swift [hunk 2]
```diff
@@ -19,10 +19,11 @@ class MiniSDCardMeasurementsParser: SDMeasurementsParser {
                 if measurementInfo.count == firstLineColumns,
                    let uuidString = getUUID(lineString: lineString) {
                     lastKnownUUID = SessionUUID(rawValue: uuidString)
+                } else {
```
Comment (mpanuszewska) [inline AirCasting/SDSync/Parsing/MiniSDCardMeasurementsParser.swift:22]: This is a fix for a misleading error message, we only want to process the line if it's not a "first line", which contains UUID

File: AirCasting/SDSync/SDSyncFileValidationService.swift [hunk 3]
```diff
@@ -95,7 +95,7 @@ struct SDSyncFileValidationService: SDSyncFileValidator {
     
     private func lineIsCorrupted(_ line: String) -> Bool {
         let fields = line.split(separator: ",")
-        return !line.isEmpty && fields.count != expectedFieldsCount
+        return !line.isEmpty && fields.count != expectedFieldsCount && fields.count != 1
```
Comment (mpanuszewska) [inline AirCasting/SDSync/SDSyncFileValidationService.swift:98]: We expect a line to have only one value: UUID, so we want to reflect that in stats. I guess in some cases the data may be corrupted in a way that it leaves only one line AND it's not UUID but it wouldn't be super easy to implement and it's such an edge case that I don't mind leaving that as is.

File: AirCasting/CoreData/MeasurementStream.swift [hunk 4]
```diff
@@ -9,6 +9,8 @@ enum MeasurementStreamSensorName: String {
     case pm1 = "AirBeam3-PM1"
     case pm2_5 = "AirBeam3-PM2.5"
     case pm10 = "AirBeam3-PM10"
```
Comment (jankrzempek) [inline AirCasting/CoreData/MeasurementStream.swift]: if we name mini, starting with `mini` shouldn't we be naming ab3 related sensors with ex. `ab3_pm10`?

File: AirCasting/SDSync/MobileSessions/SDCardMobileSessionsSavingService.swift [hunk 5]
```diff
@@ -95,7 +95,8 @@ class SDCardMobileSessionsSavingService: SDCardMobileSessionssSaver {
         Log.info("Processing mobile session: \(self.sessionUUID)")
         
         guard let processedSession = getSessionData(sessionUUID: sessionUUID, deviceID: deviceID) else {
-            Log.info("Ignoring session \(self.sessionUUID). Moving forward.")
+
```
Comment (jankrzempek) [inline AirCasting/SDSync/MobileSessions/SDCardMobileSessionsSavingService.swift]: empty line, done with intention?

File: AirCasting/SDSync/MobileSessions/SDCardMobileSessionsSavingService.swift [hunk 6]
```diff
@@ -197,16 +197,23 @@ class SDCardMobileSessionsSavingService: SDCardMobileSessionssSaver {
...
-        streamsWithMeasurements[SDStream(sessionUUID: measurements.sessionUUID, deviceID: deviceID, name: .pm2_5, header: .pm2_5), default: []].append(SDSyncMeasurement(measuredAt: measurements.date, value: measurements.pm2_5, location: location))
-        if let pm10 = measurements.pm10 {
-            streamsWithMeasurements[SDStream(sessionUUID: measurements.sessionUUID, deviceID: deviceID, name: .pm10, header: .pm10), default: []].append(SDSyncMeasurement(measuredAt: measurements.date, value: pm10, location: location))
+        
+        if deviceID.starts(with: "AirBeamMini") {
+            streamsWithMeasurements[SDStream(sessionUUID: measurements.sessionUUID, deviceID: deviceID, name: .mini_pm1, header: .pm1), default: []].append(SDSyncMeasurement(measuredAt: measurements.date, value: measurements.pm1, location: location))
+            streamsWithMeasurements[SDStream(sessionUUID: measurements.sessionUUID, deviceID: deviceID, name: .mini_pm2_5, header: .pm2_5), default: []].append(SDSyncMeasurement(measuredAt: measurements.date, value: measurements.pm2_5, location: location))
+        } else {
+
```
Comment (jankrzempek) [inline AirCasting/SDSync/MobileSessions/SDCardMobileSessionsSavingService.swift]: empty line

**Review by jankrzempek — APPROVED** [review]
Testes on AB3 - working well, great job 🚀

File: AirCasting/CoreData/MeasurementStream.swift (hunk 4 above)
Comment (mpanuszewska) [reply AirCasting/CoreData/MeasurementStream.swift]: Yeah, makes sense 👌

File: AirCasting/SDSync/MobileSessions/SDCardMobileSessionsSavingService.swift (hunk 5 above)
Comment (mpanuszewska) [reply AirCasting/SDSync/MobileSessions/SDCardMobileSessionsSavingService.swift]: I am a little spoiled with automatic code formatting tools, maybe I'll add some to our git pipeline if I have a spare minute.

**Pull request merged**